### DISCUSSION TITLE
Is it possible to use GQL to query entries by the site *language* or query for a list of all sites?

### DISCUSSION THREAD

**Discussion by weotch** [craftcms/cms#14631] [Q&A] [answered]
We want to make GQL queries that select entries by their Site's language rather than by the site handle or id.  Is this possible?  

If not, is there a way to query for a list of all Sites, returning their id, handle, and language codes?

And if _that's_ not possible, can you recommend how we can inject some transforming PHP that lives upstream of the Craft query builder and maps incoming ISO language codes to Site handles?

Thanks!

**Comment by brandonkelly** [comment, marked as answer]
Good question! Neither of those are possible yet, but it does seem like you should be able to query for elements by their language.

So I just added a `language` element query param (and corresponding GraphQL query argument) for Craft 4.9 and 5.0 (dcbd9a6803c992b076bcaa76d5222673042108bf).

If you need this now, you can update to the `4.9` branch by changing your `craftcms/cms` requirement in `composer.json` to `4.9.x-dev` and run `composer update`.

**Reply by weotch** [reply]
Thank you!

**Reply by brandonkelly** [reply]
Craft 4.9 is out now with that change.
